modules/c++/xml.lite/unittests/test_Speed.cpp

```diff
@@ -0,0 +1,413 @@
+/* =========================================================================
+ * This file is part of xml.lite-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2014, MDA Information Systems LLC
+ *
+ * xml.lite-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <string>
+#include <stdio.h>
+#include <import/io.h>
+#include <xml/lite/Speed.h>
+#include "TestCase.h"
+
+
+//#define __TEST_XML_LITE_SPEED_DEBUG__
+
+std::string writexml(const xml::lite::Speed &xml)
+{
+  io::StringStream ss_buf;
+  xml.getRootElement()->print(ss_buf);
+
+  std::string out = ss_buf.stream().str();
+#ifdef  __TEST_XML_LITE_SPEED_DEBUG__
+  std::cout<<out<<std::endl;
+#endif
+
+  return out;
+}
+
+// Start with an empty root node called "root"
+#define SETUP_XML  xml::lite::Speed xml; xml.initRoot("root");
```

Comment:
The alternate constructor could make sense. I'm going to defer creating it since, that might encourage adding extra complexity, e.g. setting root attributes, namespaces, basic document structure, etc. in the constructor. Whether that functionality (and with it, complexity) is wanted can be decided at a later time, at the risk of that time never coming.